### PR TITLE
Add class to align inline submit button with fields that don't have t…

### PR DESCRIPTION
**Pull request by LauraSteiner**
Add class to align inline submit button with fields that don't have top labels

I couldn't think of a good class name for this.  I'd appreciate suggestions!

### CODE REVIEW

File: css/_single_theme.css.php [hunk 1]
```diff
@@ -1115,6 +1121,12 @@
 		width:100%<?php echo esc_html( $important ) ?>;
 	}
 
+	.<?php echo esc_html( $style_class ) ?> .frm_align_top_submit input[type=submit],
+	.<?php echo esc_html( $style_class ) ?> .frm_align_top_submit.frm_submit input[type=button],
+	.<?php echo esc_html( $style_class ) ?> .frm_align_top_submit.frm_submit button {
+		margin-top: 10px;
```
Comment (stephywells) [inline css/_single_theme.css.php]: I'm thinking this should use the number from the settings rather than hard-code it. It's called submit_margin.

File: css/_single_theme.css.php [hunk 2]
```diff
@@ -553,6 +553,12 @@
     vertical-align:middle;
 }
 
+.<?php echo esc_html( $style_class ) ?> .frm_align_top_submit input[type=submit],
```
Comment (stephywells) [inline css/_single_theme.css.php]: Since we'll be having people enter this value for now, maybe we could make the class more explanatory? Would frm_top_submit be any better?

Comment (LauraSteiner) [reply css/_single_theme.css.php]: I'm concerned that people will think "frm_top_submit" would move the submit button to the top of the form.

What do you think of some variation of frm_inline_submit, like frm_inline_submit_top?  It's not completely clear what it does, IMO, but I think it may get people in the right ballpark.

File: css/_single_theme.css.php (hunk 1 above)
Comment (LauraSteiner) [reply css/_single_theme.css.php]: Perfect.  Thank you.

**Comment by LauraSteiner** [comment]
Wonderful idea!  

With fields in a single row, checking the last would work perfectly.  If there are multiple rows, the submit button will align with the top row.  What do you think about checking the first field instead of the last?

**Comment by LauraSteiner** [comment]
I've made the changes based on your awesome suggestions.  Making the inline submit setting useful for people is exciting!  Looking forward to your comments.

File: classes/helpers/FrmFormsHelper.php [hunk 3]
```diff
@@ -267,7 +267,7 @@ public static function get_default_html( $loc ) {
 		if ( $loc == 'submit' ) {
             $draft_link = self::get_draft_link();
             $default_html = <<<SUBMIT_HTML
-<div class="frm_submit">
+<div class="frm_submit [inline_submit]">
```
Comment (stephywells) [inline classes/helpers/FrmFormsHelper.php]: Can we revert this change and add the class with frm_inline_form in the form container instead? Since we're adding the styling, we don't really need an extra class for this so we can use ".frm_inline_form .frm_submit button" instead. It gets complicated when we make HTML changes, so we try to avoid it when possible.

File: classes/helpers/FrmFieldsHelper.php [hunk 4]
```diff
@@ -372,6 +372,42 @@ private static function &label_position( $position, $field, $form ) {
...
+	 *
+	 * @param $field
+	 * @param $form
+	 *
+	 * @return bool
+	 */
+	private static function field_has_top_label( $field, $form ) {
+
+		$label_pos = self::label_position( $field['label'], $field, $form );
+		if ( 'top' == $label_pos ) {
```
Comment (stephywells) [inline classes/helpers/FrmFieldsHelper.php]: There are a few label positions that would need this. Hidden and inside also have a space above the field.

Comment (stephywells) [reply classes/helpers/FrmFieldsHelper.php]: This line can also be simplified to something more like:
return in_array( $label_pos, array( 'top', 'inside', 'hidden' ) );

File: css/_single_theme.css.php [hunk 5]
```diff
@@ -553,6 +553,12 @@
     vertical-align:middle;
 }
 
+.<?php echo esc_html( $style_class ) ?> .frm_inline_submit_top input[type=submit],
+.<?php echo esc_html( $style_class ) ?> .frm_inline_submit_top.frm_submit input[type=button],
+.<?php echo esc_html( $style_class ) ?> .frm_inline_submit_top.frm_submit button {
+	margin-top: 0;
```
Comment (stephywells) [inline css/_single_theme.css.php]: This looks better for me with margin instead of margin-top

File: classes/helpers/FrmFormsHelper.php (hunk 3 above)
Comment (LauraSteiner) [reply classes/helpers/FrmFormsHelper.php]: Happy to set this up as you'd like!  I'm not completely sure what you're suggesting here.  Would the extra class be added to the form container?  So the two options would be something like:

.frm_inline_form.frm_form_inline_submit  .form_submit button  
.frm_inline_form.frm_form_inline_submit_top .form_submit button

Or somewhere else?

File: css/_single_theme.css.php (hunk 5 above)
Comment (LauraSteiner) [reply css/_single_theme.css.php]: Sounds good.

File: classes/helpers/FrmFormsHelper.php (hunk 3 above)
Comment (LauraSteiner) [reply classes/helpers/FrmFormsHelper.php]: Moving the styling to the form container is so much cleaner and more elegant.  Thank you for this wonderful idea!

File: classes/helpers/FrmFormsHelper.php [hunk 6]
```diff
@@ -291,8 +291,34 @@ public static function get_draft_link() {
...
+				$inline_class = 'frm_inline_submit ';
+			} else {
+				$inline_class = 'frm_inline_submit_top ';
+			}
+		}
+
+		return str_replace( '[inline_submit]', $inline_class, $button_html );
+	}
+
+	public static function get_custom_submit( $html, $form, $submit, $form_action, $values, $form_has_top_labels ) {
```
[the remark this answers is not in the document]
Comment (LauraSteiner) [reply classes/helpers/FrmFormsHelper.php]: Great thinking!  With your awesome idea of moving the styling to the form container, I no longer need to interfere with the submit button HTML at all.  So much nicer this way!

File: classes/helpers/FrmFormsHelper.php [hunk 7]
```diff
@@ -291,6 +291,9 @@ public static function get_draft_link() {
         return $link;
     }
 
+
+
+
```
Comment (stephywells) [inline classes/helpers/FrmFormsHelper.php]: Looks like there is some extra space here that needs to be removed.

File: classes/helpers/FrmFieldsHelper.php [hunk 8]
```diff
@@ -372,6 +372,58 @@ private static function &label_position( $position, $field, $form ) {
...
+	 *
+	 * @return string
+	 */
+	public static function maybe_return_inline_submit_top_class( $form ) {
+
+		if ( self::form_has_top_labels( $form ) ) {
+			return ' frm_inline_submit_top';
+		}
+
+		return '';
```
Comment (stephywells) [inline classes/helpers/FrmFieldsHelper.php]: Does this feel any cleaner to you?

return self::form_has_top_labels( $form ) ? ' frm_inline_submit_top' : '';

File: classes/helpers/FrmFormsHelper.php [hunk 9]
```diff
@@ -481,10 +499,12 @@ public static function get_form_style_class( $form = false ) {
 			$form = $form->options;
 		}
 
+
```
Comment (stephywells) [inline classes/helpers/FrmFormsHelper.php]: Do you see the Scrutinizer notices when you view this page?
https://github.com/Strategy11/formidable-forms/pull/73/files

Looks like there's an extra line here too.

File: classes/helpers/FrmFieldsHelper.php [hunk 10]
```diff
@@ -372,6 +372,58 @@ private static function &label_position( $position, $field, $form ) {
...
+	 */
+
+	private static function form_has_top_labels( $form ) {
+		$fields = $form['fields'];
+
+		if ( count( $fields ) <= 0 ) {
+			return false;
+		}
+
+		return self::field_has_top_label( reset( $fields ), $form );
```
Comment (stephywells) [inline classes/helpers/FrmFieldsHelper.php]: This is checking the label position for the first field. Do you think it would be any better to check the last field instead? Which way would be most reliable?

File: css/_single_theme.css.php [hunk 11]
```diff
@@ -553,6 +556,12 @@
     vertical-align:middle;
 }
 
+.frm_inline_form.<?php echo esc_html( $style_class ) ?> input[type=submit],
+.frm_inline_form.<?php echo esc_html( $style_class ) ?> .frm_submit input[type=button],
+.frm_inline_form.<?php echo esc_html( $style_class ) ?> .frm_submit button{
+	margin: 0 <?php echo esc_html( $important ) ?>;
+}
```
Comment (stephywells) [inline css/_single_theme.css.php]: Maybe we should combine this section with the one at line 508?
https://github.com/Strategy11/formidable-forms/pull/73/files#diff-a1eccabbb05681e6395f97699a181f58R509

File: css/_single_theme.css.php [hunk 12]
```diff
@@ -504,6 +505,8 @@
 	visibility:hidden;
 }
 
+.frm_inline_form.frm_inline_submit_top.<?php echo esc_html( $style_class ) ?> .frm_submit input,
+.frm_inline_form.frm_inline_submit_top.<?php echo esc_html( $style_class ) ?> .frm_submit button,
 .<?php echo esc_html( $style_class ) ?> .frm_submit.frm_inline_submit input,
 .<?php echo esc_html( $style_class ) ?> .frm_submit.frm_inline_submit button {
 	margin-top: 0 <?php echo esc_html( $important ) ?>;
```
Comment (stephywells) [inline css/_single_theme.css.php]: This works better for me with margin:0 instead of margin-top:0

**Comment by stephywells** [comment]
These changes have been added into 3.0, so I won't merge this one. But it's there. Thanks for all your work on this Laura!
